### PR TITLE
Updated RT goal handle to handle cancel requests

### PR DESCRIPTION
**Pull request by nlamprian**
With the proposed changes, the real time goal handle allows for a cancel request to happen and the action to be terminated at a later time from within the real time thread.

### CODE REVIEW

File: include/realtime_tools/realtime_server_goal_handle.h [hunk 1]
```diff
@@ -55,6 +55,7 @@ class RealtimeServerGoalHandle
   uint8_t state_;
 
   bool req_abort_;
+  bool req_cancel_;
```
Comment (ipa-mdl) [inline include/realtime_tools/realtime_server_goal_handle.h]: needs to be initialized in constructor

**Comment by ipa-mdl** [comment]
This might help to fix some problems with canceling in `joint_trajectory_controller`, perhaps even https://github.com/ros-controls/ros_controllers/issues/48

**Comment by bmagyar** [comment]
I'm not sure what is the best way to force travis to re-parse the CI configuration... Could you please rebase the PR on the latest `kinetic-devel`?

**Comment by bmagyar** [comment]
Great, thank you! :+1:

**Comment by bmagyar** [comment]
Should we cherry-pick this back into Indigo?

**Pull request merged**